src/Neo/SmartContract/Manifest/ContractAbi.cs

```diff
@@ -25,32 +27,51 @@ namespace Neo.SmartContract.Manifest
     /// <remarks>For more details, see NEP-14.</remarks>
     public class ContractAbi : IInteroperable
     {
-        private IReadOnlyDictionary<(string, int), ContractMethodDescriptor> methodDictionary;
+        private IReadOnlyDictionary<(string, int), ContractMethodDescriptor>? _methodDictionary;
 
         /// <summary>
         /// Gets the methods in the ABI.
         /// </summary>
-        public ContractMethodDescriptor[] Methods { get; set; }
+        public ContractMethodDescriptor[] Methods { get; set; } = [];
 
         /// <summary>
         /// Gets the events in the ABI.
         /// </summary>
-        public ContractEventDescriptor[] Events { get; set; }
+        public ContractEventDescriptor[] Events { get; set; } = [];
+
+        /// <summary>
+        /// An object with each member having a name (a string consisting of one or more identifiers joined by dots) and a value of ExtendedType object.
+        /// </summary>
+        public Dictionary<string, ExtendedType>? NamedTypes { get; set; }
 
         void IInteroperable.FromStackItem(StackItem stackItem)
         {
-            Struct @struct = (Struct)stackItem;
-            Methods = ((Array)@struct[0]).Select(p => p.ToInteroperable<ContractMethodDescriptor>()).ToArray();
-            Events = ((Array)@struct[1]).Select(p => p.ToInteroperable<ContractEventDescriptor>()).ToArray();
+            var data = (Struct)stackItem;
+            Methods = [.. ((Array)data[0]).Select(p => p.ToInteroperable<ContractMethodDescriptor>())];
+            Events = [.. ((Array)data[1]).Select(p => p.ToInteroperable<ContractEventDescriptor>())];
+
+            if (data.Count >= 3 && !data[2].IsNull)
+                NamedTypes = ((Map)data[2]).ToDictionary(p => p.Key.GetString()!, p => p.Value.ToInteroperable<ExtendedType>());
+            else
+                NamedTypes = null;
+
+            ValidateExtendedTypes();
         }
 
         public StackItem ToStackItem(IReferenceCounter referenceCounter)
         {
-            return new Struct(referenceCounter)
+            var ret = new Struct(referenceCounter)
             {
                 new Array(referenceCounter, Methods.Select(p => p.ToStackItem(referenceCounter))),
-                new Array(referenceCounter, Events.Select(p => p.ToStackItem(referenceCounter))),
+                new Array(referenceCounter, Events.Select(p => p.ToStackItem(referenceCounter)))
             };
+
+            if (NamedTypes != null)
+            {
+                ret.Add(new Map(NamedTypes.ToDictionary(p => (PrimitiveType)p.Key, p => (StackItem)p.Value.ToStackItem(referenceCounter)), referenceCounter));
+            }
```

Comment:
I have a concern: manifest serialization format doesn't depend on hardforks, and we don't change native Management in any way. It means that there's a tiny possibility of a statediff during an update to this version: if someone deploys contract with extended types, then deployment transaction will fail on old nodes and will succeed on updated nodes.

This fact may probably be ignored, but at least we should know about it.